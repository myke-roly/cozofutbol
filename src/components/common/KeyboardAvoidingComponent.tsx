import React, { useState } from 'react'
import {
  View,
  KeyboardAvoidingView,
  Dimensions,
  Platform,
  KeyboardAvoidingViewProps,
} from 'react-native'

const windowHeight = Dimensions.get('window').height

// TODO: CREATE DINAMIC SCROLL VIEW TO SET HEIGHT ON KEYBOARD OPEN AND CLOSE

const KeyboardAvoidingComponent = ({
  keyboardVerticalOffset,
  ...props
}: KeyboardAvoidingViewProps) => {
  const [verticalOffset, setVerticalOffset] = useState<number | undefined>(
    keyboardVerticalOffset,
  )

  return (
    <View
      onLayout={({ nativeEvent: { layout } }) =>
        setVerticalOffset(windowHeight - layout.height)
      }>
      <KeyboardAvoidingView
        behavior={Platform.select({ ios: 'padding' })}
        keyboardVerticalOffset={verticalOffset}
        {...props}
      />
    </View>
  )
}

export default KeyboardAvoidingComponent
