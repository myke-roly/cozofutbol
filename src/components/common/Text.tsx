import React from 'react'
import {
  Text as RNText,
  TextProps as RNTextProps,
  StyleSheet,
} from 'react-native'

type TextProps = {
  level?: '1' | '2' | '3' | '4' | '5'
  strong?: boolean
  centered?: boolean
} & RNTextProps

const Text = ({ level = '4', strong, centered, children }: TextProps) => {
  const fontSizeStyle = {
    '1': styles.level1,
    '2': styles.level2,
    '3': styles.level3,
    '4': styles.level4,
    '5': styles.level5,
  }

  const strongStyle = strong && styles.strong
  const centeredStyle = centered && styles.centered

  const finalStyles = {
    ...fontSizeStyle[level],
    ...strongStyle,
    ...centeredStyle,
  }

  return <RNText style={finalStyles}>{children}</RNText>
}

const styles = StyleSheet.create({
  level1: {
    fontSize: 10,
  },
  level2: {
    fontSize: 12,
  },
  level3: {
    fontSize: 14,
  },
  level4: {
    fontSize: 16,
  },
  level5: {
    fontSize: 20,
  },
  strong: {
    fontWeight: 'bold',
  },
  centered: {
    textAlign: 'center',
  },
})

export default Text
