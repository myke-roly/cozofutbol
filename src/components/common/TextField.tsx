import React from 'react'
import {
  KeyboardTypeOptions,
  StyleSheet,
  Text,
  TextInput,
  TextInputProps,
  View,
} from 'react-native'

type TextFieldProps = {
  value: string
  onChange: (value: string) => void
  placeholder: string
  label: string
  type?: KeyboardTypeOptions
} & TextInputProps

const TextField = ({
  placeholder,
  onChange,
  value,
  label,
  type = 'default',
  ...props
}: TextFieldProps) => {
  return (
    <View>
      <Text>{label}</Text>
      <TextInput
        style={styles.input}
        {...props}
        placeholder={placeholder}
        keyboardType={type}
        value={value}
        onChangeText={onChange}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: 'gray',
    padding: 8,
    borderRadius: 8,
  },
})

export default TextField
