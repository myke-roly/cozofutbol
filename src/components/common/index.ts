export { default as TextField } from '../ui/TextField'
export { default as Text } from './Text'
export { default as KeyboardAvoidingComponent } from './KeyboardAvoidingComponent'
