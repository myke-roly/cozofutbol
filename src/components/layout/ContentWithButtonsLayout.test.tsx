import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'
import { ContentWithButtonsLayout } from './'
import { View } from 'react-native'
import { Text } from '../common'

describe('ContentWithButtonsLayout', () => {
  const fn = jest.fn()

  it('should render correctly', () => {
    const tree = renderer.create(<ContentWithButtonsLayout />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with children', () => {
    const tree = renderer
      .create(
        <ContentWithButtonsLayout>
          <View>
            <Text>Child Content 1</Text>
          </View>
        </ContentWithButtonsLayout>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with primary button', () => {
    const tree = renderer
      .create(
        <ContentWithButtonsLayout
          primaryButton={{
            title: 'primary-title-test',
            onPress: fn,
          }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with secondary button', () => {
    const tree = renderer
      .create(
        <ContentWithButtonsLayout
          secondaryButton={{
            title: 'secondary-title-test',
            onPress: fn,
          }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with primary and secondary button', () => {
    const tree = renderer
      .create(
        <ContentWithButtonsLayout
          primaryButton={{
            title: 'primary-title-test',
            onPress: fn,
          }}
          secondaryButton={{
            title: 'secondary-title-test',
            onPress: fn,
          }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly with secondary button outline variant', () => {
    const tree = renderer
      .create(
        <ContentWithButtonsLayout
          primaryButton={{
            title: 'primary-title-test',
            onPress: fn,
            variant: 'outline',
          }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly if isfocued is true', () => {
    const tree = renderer
      .create(
        <ContentWithButtonsLayout
          primaryButton={{
            title: 'primary-title-test',
            onPress: fn,
          }}
          isfocued={true}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly onLayout sets the buttons height', () => {
    const mockNativeEvent = {
      nativeEvent: {
        layout: {
          height: 100,
        },
      },
    }

    const { getByTestId, toJSON } = render(
      <ContentWithButtonsLayout
        primaryButton={{
          title: 'primary-title-test',
          onPress: fn,
        }}
      />,
    )

    fireEvent(getByTestId('buttons-view'), 'layout', mockNativeEvent)
    expect(toJSON()).toMatchSnapshot()
  })
})
