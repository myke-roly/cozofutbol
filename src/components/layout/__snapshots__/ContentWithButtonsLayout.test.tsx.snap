// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentWithButtonsLayout Should render correctly if isfocued is true 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 0,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View />
  </RCTScrollView>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#fafafa",
          "bottom": 0,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
          "position": "absolute",
          "width": "100%",
        },
        {
          "elevation": 10,
          "shadowColor": "#232323",
          "shadowOffset": {
            "height": 10,
            "width": 0,
          },
        },
      ]
    }
    testID="buttons-view"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="btn-presable-view"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 8,
            "borderWidth": 1,
            "elevation": 5,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
            "shadowColor": "#232323",
            "shadowOffset": {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fafafa",
              "fontSize": 16,
            }
          }
        >
          primary-title-test
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ContentWithButtonsLayout Should render correctly onLayout sets the buttons height 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 100,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View />
  </RCTScrollView>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#fafafa",
          "bottom": 0,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
          "position": "absolute",
          "width": "100%",
        },
        false,
      ]
    }
    testID="buttons-view"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="btn-presable-view"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 8,
            "borderWidth": 1,
            "elevation": 5,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
            "shadowColor": "#232323",
            "shadowOffset": {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fafafa",
              "fontSize": 16,
            }
          }
        >
          primary-title-test
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ContentWithButtonsLayout Should render correctly with secondary button outline variant 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 0,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View />
  </RCTScrollView>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#fafafa",
          "bottom": 0,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
          "position": "absolute",
          "width": "100%",
        },
        false,
      ]
    }
    testID="buttons-view"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="btn-presable-view"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#232323",
            "borderRadius": 8,
            "borderWidth": 1,
            "elevation": 0,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
            "shadowColor": "#232323",
            "shadowOffset": {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#232323",
              "fontSize": 16,
            }
          }
        >
          primary-title-test
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ContentWithButtonsLayout should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 0,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`ContentWithButtonsLayout should render correctly with children 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 0,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View>
        <Text
          style={
            [
              undefined,
              {
                "fontSize": 16,
              },
            ]
          }
        >
          Child Content 1
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`ContentWithButtonsLayout should render correctly with primary and secondary button 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 0,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View />
  </RCTScrollView>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#fafafa",
          "bottom": 0,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
          "position": "absolute",
          "width": "100%",
        },
        false,
      ]
    }
    testID="buttons-view"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="btn-presable-view"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 8,
            "borderWidth": 1,
            "elevation": 5,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
            "shadowColor": "#232323",
            "shadowOffset": {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fafafa",
              "fontSize": 16,
            }
          }
        >
          primary-title-test
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="btn-presable-view"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 8,
            "borderWidth": 0,
            "elevation": 0,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
            "shadowColor": "#232323",
            "shadowOffset": {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#232323",
              "fontSize": 16,
            }
          }
        >
          secondary-title-test
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ContentWithButtonsLayout should render correctly with primary button 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 0,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View />
  </RCTScrollView>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#fafafa",
          "bottom": 0,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
          "position": "absolute",
          "width": "100%",
        },
        false,
      ]
    }
    testID="buttons-view"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="btn-presable-view"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 8,
            "borderWidth": 1,
            "elevation": 5,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
            "shadowColor": "#232323",
            "shadowOffset": {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fafafa",
              "fontSize": 16,
            }
          }
        >
          primary-title-test
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ContentWithButtonsLayout should render correctly with secondary button 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "marginVertical": 24,
          "width": "90%",
        },
        {
          "marginBottom": 0,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
  >
    <View />
  </RCTScrollView>
</View>
`;
